### PR TITLE
KK-220 | Add docker and gitlab pipeline configs

### PR DESCRIPTION
**Pull request by tuomas777**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  app:
+    container_name: kukkuu-admin
+    build:
+      context: .
+      target: ${DOCKER_TARGET:-development}
+    volumes:
+      - '.:/app'
+      - '/app/node_modules'
+    ports:
+      - '3000:3000'
```
Comment (badrange) [inline docker-compose.yml]: By the way: It is likely that a developer wants to run both kukkuu-ui and kukkuu-admin on the same computer. Maybe this one should run on 3001 or something?

Comment (tuomas777) [reply docker-compose.yml]: Yeah good point, I will change that.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    refs:
+      - develop
+
+build-production:
+  extends: .build
+  variables:
+    DOCKER_IMAGE_NAME: '$CI_PROJECT_NAME-production'
+  only:
+    refs:
+      - master
```
Comment (antti-mikael) [inline .gitlab-ci.yml]: There is no need to define multiple builds if all the build jobs end up building the same image (i.e. no build-time arguments are passed to `docker build`).

File: .env [hunk 3]
```diff
@@ -0,0 +1 @@
+PORT=3001
```
Comment (antti-mikael) [inline .env]: Usually the `.env` file is not committed but a `.env.example` that can be copied as `.env`.

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+USER root
+
+RUN apt-install.sh build-essential
+
+USER appuser
+RUN yarn && yarn cache clean --force
+
+USER root
+RUN apt-cleanup.sh build-essential
```
Comment (antti-mikael) [inline Dockerfile]: Removing stuff in a separate layer does not make the image smaller. There two possible solutions:

1. Do everything in a single layer
```Dockerfile
USER root

RUN apt-install.sh build-essential \
    && su -c "yarn && yarn clean cache --force" appuser \
    && apt-cleanup.sh build-essential
```
2. Use a separate stage
```Dockerfile
FROM helsinkitest/node:12-slim as appbuilder

RUN apt-install.sh build-essential
COPY package.json yarn.lock ./
RUN yarn

FROM helsinkitest/node:12-slim as appbase

COPY --from=appbuilder --chown=appuser:appuser <SRC> <DST>
```

File: .env (hunk 3 above)
Comment (tuomas777) [reply .env]: Is that actually the situation in these React apps as well, at least in [create-react-app docs](https://create-react-app.dev/docs/adding-custom-environment-variables/#adding-development-environment-variables-in-env) they advice the other way.

File: .gitlab-ci.yml (hunk 2 above)
Comment (tuomas777) [reply .gitlab-ci.yml]: Ok! But actually I will be adding a couple of `DOCKER_BUILD_ARG_*`s to those in the next PR I create, so if those multiple builds are then needed anyway, I think I don't bother to change that back and forth :)

File: Dockerfile (hunk 4 above)
Comment (tuomas777) [reply Dockerfile]: Ok thanks, I'll go with the first one.

Comment (tuomas777) [reply Dockerfile]: Made the change. Impact was pretty nice, size went from 895MB to 736MB.

**Pull request merged**